apps/web/src/components/conditions/Conditions.tsx

```diff
@@ -108,92 +119,100 @@ export function Conditions({
-      })}
+                        </Wrapper>
+                      ) : (
+                        <Text ml={14} color={colors.B60}>
+                          When
+                        </Text>
+                      )}
+                    </Grid.Col>
+                    <Grid.Col span={3}>
+                      <Controller
+                        control={control}
+                        name={`conditions.0.children.${index}.on`}
+                        defaultValue={defaultOnFilter}
+                        render={({ field }) => {
+                          return (
+                            <Select
+                              placeholder="On"
+                              data={FilterPartTypeList}
+                              {...field}
+                              onChange={handleOnChildOnChange(index)}
+                              data-test-id="conditions-form-on"
+                            />
+                          );
+                        }}
+                      />
+                    </Grid.Col>
+                    <When truthy={filterFieldOn === FilterPartTypeEnum.WEBHOOK}>
+                      <WebHookUrlForm control={control} index={index} />
+                      <EqualityForm webhook control={control} index={index} />
+                    </When>
+                    <When truthy={filterFieldOn === FilterPartTypeEnum.PREVIOUS_STEP}>
+                      <PreviousStepsConditionRow customData={customData} control={control} index={index} />
+                    </When>
+                    <When
+                      truthy={[FilterPartTypeEnum.IS_ONLINE, FilterPartTypeEnum.IS_ONLINE_IN_LAST].includes(
+                        filterFieldOn
+                      )}
+                    >
+                      <OnlineConditionRow fieldOn={filterFieldOn} control={control} index={index} />
+                    </When>
+                    <When
+                      truthy={[
+                        FilterPartTypeEnum.PAYLOAD,
+                        FilterPartTypeEnum.SUBSCRIBER,
+                        FilterPartTypeEnum.TENANT,
+                      ].includes(filterFieldOn)}
+                    >
+                      <EqualityForm customData={customData} control={control} index={index} />
+                    </When>
+                  </Grid>
+                </Grid.Col>
+                <ConditionRowMenu onDuplicate={() => handleDuplicate(index)} onDelete={() => handleDelete(index)} />
+              </Grid>
+            </div>
+          );
+        })}
+      </div>
 
       <Group position="left">
         <Button
           variant="outline"
           onClick={() => {
             append({
               operator: 'EQUAL',
-              on: FilterPartTypeEnum.TENANT,
-              field: 'identifier',
-              value: '',
+              // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+              // @ts-ignore
+              on: defaultOnFilter,
```

Comment:
I am not sure how to fix the typescript error here. It expects on to be of type `FilterPartTypeEnum.PAYLOAD | FilterPartTypeEnum.SUBSCRIBER`

Reply:
yes, because if you set `defaultOnFilter` to let's say `FilterPartTypeEnum.PREVIOUS_STEP` then the form won't be if the correct state as for that type other fields are required... 

I feel that we should call here a function that will return an empty state for the row depending on the `FilterPartTypeEnum`, ex.

```
const getConditionRow = (filterType: FilterPartTypeEnum) => {
   switch (filterType) {
      case FilterPartTypeEnum.SUBSCRIBER:
         return { ... }
      case FilterPartTypeEnum.TENANT:
         return { ... }
   }
}


...
onClick={() => {
    append(getConditionRow(defaultOnFilter));
}}
```